src/wp-admin/includes/class-wp-plugins-list-table.php

```diff
@@ -292,11 +292,20 @@ public function prepare_items() {
 			}
 		}
 
-		if ( strlen( $s ) ) {
+		if ( isset( $s ) && strlen( $s ) ) {
```

Comment:
This change doesn't appear to be related to the addition of the new filter. @nate-allen Was this part of a patch for another ticket?

Reply:
Hi @costdev. It's required for the unit test for this ticket.

Reply:
Pinging @TimothyBJacobs and @spacedmonkey as REST API maintainers. Are you happy for this source change to be made here to allow PHPUnit tests to pass, or should it be made in a separate ticket, or should the test(s) instead set a value for the `$s` global?

Reply:
@costdev This doesn't look REST API related to me? But since I'm here 🙂.

To me it looks like the relevant globals should be getting initialized in the test. Though I think @SergeyBiryukov or @hellofromtonya could give a more informed opinion on the direction for these kinds of unit tests.

Reply:
Thanks @TimothyBJacobs! It appears my brain skipped a beat when I posted that at 3AM 😅

Yeah that's what I was thinking - and that the `isset( $s )` check, if it's going to be added at some point, should probably be separate rather than bundled in the addition of a new filter.

Let's see what Sergey or Tonya think too 🙂

Reply:
Thanks for ping @TimothyBJacobs and @costdev.

>To me it looks like the relevant globals should be getting initialized in the test.

In this PR, yes, I agree. 

How to do it?

- Store each global's original value in a new property added to the test class. 
    - Add the property.
    - The initialization of the property in either the `before_set_up_class()` or `set_up()` fixture method.
- The test is then able to modify the global to its needs.
-  How to restore / reset the original value? In the `tear_down()`, set the global to its the original value using the property. 
    - Why the `tear_down()` and not within the test method? The `tear_down()` ensures the global gets reset, as it runs after each test regardless if the test fails and skips any of it test code.

[Here's an example test](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/webfonts/wpThemeJsonWebfontsHandler.php).

Reply:
Thank you @hellofromtonya (and everyone else that has helped move this along!) I will make these changes and reach out if I have any issues.